client/reader/discover/discover-stream.js

```diff
@@ -91,6 +93,18 @@ const DiscoverStream = ( props ) => {
 		selectedStreamName: selectedTab,
 	};
 
+	if ( selectedTab === 'add-new' ) {
+		return (
+			<ReaderMain className={ clsx( 'following main', props.className ) }>
+				<DiscoverHeader selectedTab={ effectiveTabSelection } width={ props.width } />
+				<DiscoverNavigationV2 selectedTab={ selectedTab } />
```

Comment:
We are using `DiscoverHeader` and `DiscoverNavigationV2` twice, better if we can use them only once.

```suggestion
				<HeaderAndNavigation />
```

Implementation of `HeaderAndNavigation`:

```
	const HeaderAndNavigation = () => {
		return (
			<>
				<DiscoverHeader selectedTab={ effectiveTabSelection } width={ props.width } />
				{ isDiscoveryV2Enabled() ? (
					<DiscoverNavigationV2 selectedTab={ selectedTab } />
				) : (
					<DiscoverNavigation width={ props.width } selectedTab={ selectedTab } />
				) }

				{ selectedTab === 'tags' && (
					<DiscoverTagsNavigation
						width={ props.width }
						selectedTag={ selectedTag }
						onTagSelect={ handleTagSelect }
					/>
				) }
			</>
		);
	};
```

This will also simplify the stream component as following:

```
		<Stream { ...streamProps }>
			<HeaderAndNavigation />
		</Stream>
```

Reply:
That looks good to me! Done in: aeb0b8f18555a030d0f4a198ae9eee527b1eef52